frontend/src/scenes/surveys/surveyViewViz.tsx

```diff
@@ -53,99 +54,70 @@ const formatCount = (count: number, total: number): string => {
-                                },
-                            },
-                            {
-                                count: sent,
-                                label: 'Sent',
-                                classes: `rounded-r ${seen === 0 && dismissed === 0 ? 'rounded-l' : ''}`,
-                                style: {
-                                    backgroundColor: '#529B08',
-                                    width: `${sentPercentage}%`,
-                                    left: `${seenPercentage + dismissedPercentage}%`,
-                                },
-                            },
-                        ].map(({ count, label, classes, style }) => (
-                            <Tooltip
-                                key={`survey-summary-chart-${label}`}
-                                title={`${label} surveys: ${count}`}
-                                delayMs={0}
-                                placement="top"
-                            >
-                                <div
-                                    className={`h-10 text-white text-center absolute cursor-pointer ${classes}`}
-                                    // eslint-disable-next-line react/forbid-dom-props
-                                    style={style}
+                        {segments.map(({ count, label, color }, index) => {
+                            const percentage = (count / total) * 100
+                            const left = accumulatedPercentage
+                            accumulatedPercentage += percentage
+
+                            const isFirst = index === 0
+                            const isLast = index === segments.length - 1
+                            const isOnly = segments.length === 1
+
+                            return (
+                                <Tooltip
+                                    key={`stacked-bar-${label}`}
+                                    title={`${label}: ${count} (${percentage.toFixed(1)}%)`}
+                                    delayMs={0}
+                                    placement="top"
                                 >
-                                    <span className="inline-flex font-semibold max-w-full px-1 truncate leading-10">
-                                        {formatCount(count, total)}
-                                    </span>
-                                </div>
-                            </Tooltip>
-                        ))}
+                                    <div
+                                        className={clsx(
+                                            'h-10 text-white text-center absolute cursor-pointer',
+                                            isFirst || isOnly ? 'rounded-l' : '',
+                                            isLast || isOnly ? 'rounded-r' : ''
+                                        )}
+                                        // eslint-disable-next-line react/forbid-dom-props
```

Comment:
should we address https://github.com/jsx-eslint/eslint-plugin-react/blob/master/docs/rules/forbid-dom-props.md instead?